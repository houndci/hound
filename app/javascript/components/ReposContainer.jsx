import React from 'react';

import App from './ReposContainer/components/App';

export default class ReposContainer extends React.Component {
  render() {
    return(
      <App {...this.props} />
    );
  }
}

