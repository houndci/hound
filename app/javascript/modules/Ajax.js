import $ from "jquery";

export function createSubscription(subscriptionOptions) {
  return $.ajax({
    url: `/repos/${subscriptionOptions.repo_id}/subscription.json`,
    type: "POST",
    data: subscriptionOptions,
    dataType: "json"
  });
}

export function deleteSubscription(repo) {
  return $.ajax({
    url: `/repos/${repo.id}/subscription`,
    type: "DELETE",
    dataType: "json"
  });
}

export function deactivateRepo(repo) {
  return $.ajax({
    url: `/repos/${repo.id}/deactivation`,
    type: "POST",
    dataType: "text"
  });
}

export function activateRepo(repo) {
  return $.ajax({
    url: `/repos/${repo.id}/activation`,
    type: "POST",
    dataType: "text"
  });
}

export function updateOwner(id, params) {
  return $.ajax({
    url: `/owners/${id}`,
    type: "PUT",
    dataType: "json",
    data: { owner: params }
  });
}

export function upgradeSubscription(id, params) {
  return $.ajax({
    url: `/repos/${id}/subscription.json`,
    type: "PUT",
    dataType: "json",
    data: params,
    success: () => {
      document.location.href = "/repos";
    }
  });
}
