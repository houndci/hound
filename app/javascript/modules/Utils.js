export function getCSRFfromHead() {
  if (process.env.NODE_ENV === 'test') {
    return "csrf_token";
  } else {
    return document.querySelector("meta[name=csrf-token]").content;
  }
}
