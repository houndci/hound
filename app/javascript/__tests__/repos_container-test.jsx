import ReposContainer from '../components/ReposContainer/components/App';
import * as Ajax from '../modules/Ajax';

it('renders appropriately', () => {
  const wrapper = shallow(
    <ReposContainer
      authenticity_token={"csrf_token"}
      has_private_access={false}
      userHasCard={false}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('sets the filterTerm appropriately on text input', () => {
  const wrapper = mount(
    <ReposContainer
      authenticity_token={"csrf_token"}
      has_private_access={false}
      userHasCard={false}
    />
  );
  const textInput = wrapper.find('.repo-search-tools-input');
  textInput.simulate('change', {target: {value: "new text"}});

  expect(wrapper.state('filterTerm')).toBe('new text');
});

it('fetches repos and organizations on mount', () => {
  let stub = sinon.stub(ReposContainer.prototype, 'fetchReposAndOrgs');

  const wrapper = mount(
    <ReposContainer
      authenticity_token={"csrf_token"}
      has_private_access={false}
      userHasCard={false}
    />
  );

  expect(stub.callCount).toBe(1);
  stub.restore();
});
