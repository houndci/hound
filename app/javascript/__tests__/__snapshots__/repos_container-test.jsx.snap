exports[`test renders appropriately 1`] = `
<div>
  <RepoTools
    isSyncing={true}
    onPrivateClicked={[Function]}
    onRefreshClicked={[Function]}
    onSearchInput={[Function]}
    showPrivateButton={true} />
  <ReposView
    filterTerm={null}
    isProcessingId={null}
    isSyncing={true}
    onRepoClicked={[Function]}
    organizations={Array []}
    repos={Array []} />
</div>
`;
